tests/test_template.py

```diff
@@ -35,17 +36,33 @@ class TestInterferometer(BaseTest):
 
     num_subsystems = 4
 
-    def randnc(self, *arg):
-        """Normally distributed array of random complex numbers."""
-        return np.random.randn(*arg) + 1j*np.random.randn(*arg)
+    def test_integration(self):
+        """integration test for the Interferomerter."""
+        dev = qml.device('default.gaussian', wires=self.num_subsystems)
+        np.random.seed(88)
+        num_params = int(self.num_subsystems*(self.num_subsystems-1)/2)
+        theta = np.random.uniform(0, 2*np.pi, num_params)
+        phi = np.random.uniform(0, 2*np.pi, num_params)
 
-    def setUp(self):
-        super().setUp()
-        np.random.seed(35)
-        self.u1 = self.random_interferometer(self.num_subsystems)
-        self.u2 = self.random_interferometer(self.num_subsystems-1)
+        # to test whether the correct circuit is produces we inspect the
+        # Device._op_queue by patching Device.execute()
+        test = self
+        def new_execute(self, queue, expectation):
+            test.assertAllEqual([[t, p] for t, p in zip(theta, phi)], [op.parameters for op in queue])
+            test.assertAllEqual([qml.Beamsplitter]*len(theta), [type(op) for op in queue])
```

Comment:
Just trying to understand this. The Interferometer decomposition does not produce any phase rotation gates?

Reply:
nevermind, this is already being discussed above